src/components/App/App.js

```diff
@@ -60,32 +154,113 @@ function App() {
     return () => document.removeEventListener("mousedown", closePopupOnRemoteClick);
   }, []);
 
+  function handleSearchFormSubmit(searchWord) {
+    setIsSearchProcessing(true);
+    newsApi.getSearchedArticles(searchWord)
+      .then((data) => {
+        let articles = data.articles.map(article => article = { ...article, keyword: searchWord });
+        setSearchedArticles(articles);
+        localStorage.setItem("searchResults", JSON.stringify(articles));
+        setIsSearchProcessing(false);
+        setSubmitSearch(true);
+        setKeyword(searchWord);
+        localStorage.setItem("keyword", searchWord);
+      })
+      .catch((err) => console.log(err))
+      .finally(() => {
+        setIsSearchProcessing(false);
+      })
+  }
+
+  function handleArticleSave(article) {
+    const isSaved = savedArticles.find(savedArticle => savedArticle.url === article.url);
+
+    if (isSaved) {
+      mainApi
+        .deleteArticle(isSaved._id)
+        .then(() => getArticles())
```

Comment:
`        .then(() => getArticles())
`

Please, don't call `getArticles` everytime you delete an item. It's a waste of resources. You need to **filter** them